### PR TITLE
Import a file to a disk

### PR DESCRIPTION
**Pull request by jmpesp**
Use the bulk-write import functionality to upload a file to a disk, optionally making a snapshot and an image out of that snapshot. An example invocation:

    oxide disk import \
        --path crucible-tester-sparse.img \
        --threads 8 \
        --disk-name crucible-tester-disk \
        --project myproj \
        --description "testing import!" \
        --snapshot-name crucible-tester-snapshot \
        --image-name crucible-tester \
        --image-description "uploading of crucible-tester-sparse.img" \
        --image-os debian \
        --image-version 11

This will:

- create a disk named `crucible-tester-disk` in state "import-ready"
- set that disk to "importing-via-bulk-write"
- spawn 8 upload tasks and bulk write file chunks to the disk in parallel
- wait for it to be done
- stop the bulk write process
- finalize the disk, creating the snapshot `crucible-tester-snapshot`
- create the image `crucible-tester` out of that snapshot

### CODE REVIEW

**Comment by jmpesp** [comment]
Note: there's some TODO here before merging!

File: cli/src/main.rs [hunk 1]
```diff
@@ -137,7 +137,8 @@ async fn main() {
 
         // TODO this could be improved
         Some(("disk", sub_matches)) if sub_matches.subcommand_name() == Some("import") => {
-            cmd_disk::CmdDiskImport::from_arg_matches(sub_matches)
+            let real_sub_matches = sub_matches.subcommand().unwrap().1;
+            cmd_disk::CmdDiskImport::from_arg_matches(real_sub_matches)
```
Comment (ahl) [inline cli/src/main.rs:141]: oops... I need a better way of handling this generically. Thanks.

**Comment by jmpesp** [comment]
The finalize endpoint changed to using a request body instead of query params for some fields, so this PR now depends on https://github.com/oxidecomputer/oxide-sdk-and-cli/pull/96

**Comment by ahl** [comment]
@jmpesp just a heads up that I merged #96

**Comment by jmpesp** [comment]
After the repo change, this is no longer picking up pushed commits. Will close and reopen.

**Pull request closed without merging**